x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/pipeline_processors_editor_item/pipeline_processors_editor_item.tsx

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { FunctionComponent, memo, useState } from 'react';
+import {
+  EuiButtonIcon,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import { ProcessorInternal, ProcessorSelector } from '../../types';
+
+import { usePipelineProcessorsContext } from '../../context';
+
+import './pipeline_processors_editor_item.scss';
+
+import { InlineTextInput } from './inline_text_input';
+
+export interface Handlers {
+  onMove: () => void;
+  onCancelMove: () => void;
+}
+
+export interface Props {
+  processor: ProcessorInternal;
+  selected: boolean;
+  handlers: Handlers;
+  selector: ProcessorSelector;
+  description?: string;
+}
+
+export const PipelineProcessorsEditorItem: FunctionComponent<Props> = memo(
+  ({ processor, description, handlers: { onCancelMove, onMove }, selector, selected }) => {
+    const {
+      state: { editor, processorsDispatch },
+    } = usePipelineProcessorsContext();
+    const [isContextMenuOpen, setIsContextMenuOpen] = useState<boolean>(false);
+
+    const disabled = editor.mode.id !== 'idle';
+    const shouldDarkBold =
+      editor.mode.id === 'editingProcessor' ? processor.id === editor.mode.arg.processor.id : true;
```

Comment:
I wondered if this was any clearer, see what you think:

```suggestion
      editor.mode.id !== 'editingProcessor' || processor.id === editor.mode.arg.processor.id;
```